internal/pkg/cli/task_run.go

```diff
@@ -129,6 +152,61 @@ func (o *runTaskOpts) validateEnvName() error {
 	return nil
 }
 
+func (o *runTaskOpts) askTaskFamilyName() error {
+	if o.familyName != "" {
+		return nil
+	}
+
+	// TODO during Execute: list existing tasks like in ListApplications, ask whether to use existing tasks
+
+	familyName, err := o.prompt.Get(
+		fmtTaskRunFamilyNamePrompt,
+		taskRunFamilyNamePromptHelp,
+		validateTaskFamilyName,
+		prompt.WithFinalMessage("Task family name:"))
+	if err != nil {
+		return fmt.Errorf("prompt get task family name: %w", err)
+	}
+	o.familyName = familyName
+	return nil
+}
+
+func (o *runTaskOpts) askEnvName() error {
+	if o.env != "" {
+		return nil
+	}
+
+	if o.AppName() == "" {
+		o.env = envNameNone
+		return nil
+	}
+
+	existingEnv, err := o.store.ListEnvironments(o.AppName())
```

Comment:
Would it make sense to add a ListEnvironmentsWithNone method to the [ConfigSelector](https://github.com/aws/amazon-ecs-cli-v2/blob/0d26912c811d4c75d6746df2293fe753b315e911/internal/pkg/cli/selector/selector.go#L63) struct and wrap this complexity in an interface?

Reply:
I didn't do it because `ListEnvironmentsWithNone` wouldn't get used anywhere else. It'd also definitely make sense though to wrap this in an interface. In that case, would it make sense to add that method to [Select](https://github.com/aws/amazon-ecs-cli-v2/blob/0d26912c811d4c75d6746df2293fe753b315e911/internal/pkg/cli/selector/selector.go#L37)  ?

Reply:
Yes, I would add that to Select as you say. And I'd create a [new interface](https://github.com/aws/amazon-ecs-cli-v2/blob/0d26912c811d4c75d6746df2293fe753b315e911/internal/pkg/cli/interfaces.go#L335) `appEnvWithNoneSelector` specifying `configSelector` and `EnvironmentsWithNone()`.

This will let you mock calls to this method and hopefully simplify your testing.